pr-demo.js

```
@@ -3,75 +3,114 @@
+  fs.readFile('data.txt','utf8',(e,d)=>{if(e) return; cb(d)})
+}
+function fetchData(url, cb){  // overload that never gets called
+  // pretend to fetch over network
+  cb("fetched:"+url)
+}
+
+// 8. Inconsistent variable scopes & missing declarations
+for(i=0;i<3;i++){  // i is global
+   console.log(i)
+}
+if (true) {
```

Biome (1.9.4)

[error] Unexpected constant condition.



(lint/correctness/noConstantCondition)